pkg/controllers/workgenerator/controller.go

```diff
@@ -173,6 +218,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 		klog.ErrorS(updateErr, "Failed to update the resourceBinding status", "resourceBinding", bindingRef)
 		return controllerruntime.Result{}, controller.NewUpdateIgnoreConflictError(updateErr)
 	}
+	if errors.Is(syncErr, controller.ErrUserError) {
```

Comment:
what type of error is user error?  looks like we don't set the sync error in the condition when the override error !=nil? In that case, how can user know what's going on?

Reply:
for now, we'll set user error when we fail to apply the overrides or fail to extract resources from the configmap.

they can track the details in the failedManifest.